src/parameters.jl

```diff
@@ -0,0 +1,73 @@
+export AbstractParameter, AbstractSolverParameter, AbstractHyperParameter, AlgorithmicParameter
+
+export default, domain, name, granularity, nomad_type, set_default!, find, lower_bounds,
+upper_bounds, current_param_values, granularities, input_types
+
+abstract type AbstractParameter{T} end
+abstract type AbstractSolverParameter{T} <: AbstractParameter{T} end
+abstract type AbstractHyperParameter{T} <: AbstractParameter{T} end
+mutable struct AlgorithmicParameter{T, D <: AbstractDomain{T}} <: AbstractHyperParameter{T}
+  default::T
+  domain::D
+  name::String
+  granularity::Float64
+  function AlgorithmicParameter(default::T, domain::AbstractDomain{T}, name::String) where {T}
+    check_default(domain, default)
+    granularity = default_granularity(eltype(domain))
+    new{T, typeof(domain)}(default, domain, name, granularity)
+  end
+end
+default(parameter::AlgorithmicParameter{T}) where {T} = parameter.default
+domain(parameter::AlgorithmicParameter{T}) where {T} = parameter.domain
+name(parameter::AlgorithmicParameter{T}) where {T} = parameter.name
+granularity(parameter::AlgorithmicParameter{T}) where {T} = parameter.granularity
+nomad_type(::Type{T}) where {T <: Real} = "R"
+nomad_type(::Type{T}) where {T <: Integer} = "I"
+nomad_type(::Type{T}) where {T <: Bool} = "B"
+
+function check_default(domain::AbstractDomain{T}, new_value::T) where {T}
+  new_value ∈ domain || error("default value should be in domain")
+end
+
+function set_default!(parameter::AlgorithmicParameter{T}, new_value::Float64) where {T}
+  if nomad_type(eltype(domain(parameter))) == "I"
+    new_value = round(Int64, new_value)
```

Comment:
```suggestion
    new_value = round(Int, new_value)
```
On 32-bit systems, Int = Int32.